Projects/Checkers/Program.cs

```diff
@@ -0,0 +1,228 @@
+				else
+				{
+					game.PerformMove(moves[Random.Shared.Next(moves.Count)]);
+				}
+			}
+			else
+			{
+				game.PerformMove(moves[Random.Shared.Next(moves.Count)]);
+			}
+		}
+
+		RenderGameState(game, playerMoved: currentPlayer, promptPressKey: true);
+		Console.ReadKey(true);
+	}
+}
+
+void RenderGameState(Game game, Player? playerMoved = null, (int X, int Y)? selection = null, (int X, int Y)? from = null, bool promptPressKey = false)
+{
+	const char BlackPiece = '○';
+	const char BlackKing  = '☺';
+	const char WhitePiece = '◙';
+	const char WhiteKing  = '☻';
+	const char Vacant     = '·';
+
+	Console.CursorVisible = false;
+	Console.SetCursorPosition(0, 0);
+	StringBuilder sb = new();
+	sb.AppendLine();
+	sb.AppendLine("  Checkers");
+	sb.AppendLine();
+	sb.AppendLine($"    ╔═══════════════════╗");
+	sb.AppendLine($"  8 ║  {B(0, 7)} {B(1, 7)} {B(2, 7)} {B(3, 7)} {B(4, 7)} {B(5, 7)} {B(6, 7)} {B(7, 7)}  ║ {BlackPiece} = Black");
+	sb.AppendLine($"  7 ║  {B(0, 6)} {B(1, 6)} {B(2, 6)} {B(3, 6)} {B(4, 6)} {B(5, 6)} {B(6, 6)} {B(7, 6)}  ║ {BlackKing} = Black King");
+	sb.AppendLine($"  6 ║  {B(0, 5)} {B(1, 5)} {B(2, 5)} {B(3, 5)} {B(4, 5)} {B(5, 5)} {B(6, 5)} {B(7, 5)}  ║ {WhitePiece} = White");
+	sb.AppendLine($"  5 ║  {B(0, 4)} {B(1, 4)} {B(2, 4)} {B(3, 4)} {B(4, 4)} {B(5, 4)} {B(6, 4)} {B(7, 4)}  ║ {WhiteKing} = White King");
+	sb.AppendLine($"  4 ║  {B(0, 3)} {B(1, 3)} {B(2, 3)} {B(3, 3)} {B(4, 3)} {B(5, 3)} {B(6, 3)} {B(7, 3)}  ║");
+	sb.AppendLine($"  3 ║  {B(0, 2)} {B(1, 2)} {B(2, 2)} {B(3, 2)} {B(4, 2)} {B(5, 2)} {B(6, 2)} {B(7, 2)}  ║ Taken:");
+	sb.AppendLine($"  2 ║  {B(0, 1)} {B(1, 1)} {B(2, 1)} {B(3, 1)} {B(4, 1)} {B(5, 1)} {B(6, 1)} {B(7, 1)}  ║ {game.TakenCount(White),2} x {WhitePiece}");
+	sb.AppendLine($"  1 ║  {B(0, 0)} {B(1, 0)} {B(2, 0)} {B(3, 0)} {B(4, 0)} {B(5, 0)} {B(6, 0)} {B(7, 0)}  ║ {game.TakenCount(Black),2} x {BlackPiece}");
+	sb.AppendLine($"    ╚═══════════════════╝");
+	sb.AppendLine($"       A B C D E F G H");
+	sb.AppendLine();
+	if (selection is not null)
+	{
+		sb.Replace(" $ ", $"[{ToChar(game.Board[selection.Value.X, selection.Value.Y])}]");
+	}
+	if (from is not null)
+	{
+		char fromChar = ToChar(game.Board[from.Value.X, from.Value.Y]);
+		sb.Replace(" @ ", $"<{fromChar}>");
+		sb.Replace("@ ",  $"{fromChar}>");
+		sb.Replace(" @",  $"<{fromChar}");
+	}
+	PieceColor? wc = game.Winner;
+	PieceColor? mc = playerMoved?.Color;
+	PieceColor? tc = game.Turn;
+	// Note: these strings need to match in length
+	// so they overwrite each other.
+	string w = $"  *** {wc} wins ***";
+	string m = $"  {mc} moved       ";
+	string t = $"  {tc}'s turn      ";
+	sb.AppendLine(
+		game.Winner is not null ? w :
+		playerMoved is not null ? m :
+		t);
+	string p = "  Press any key to continue...";
+	string s = "                              ";
+	sb.AppendLine(promptPressKey ? p : s);
+	Console.Write(sb);
+
+	char B(int x, int y) =>
+		(x, y) == selection ? '$' :
+		(x, y) == from ? '@' :
+		ToChar(game.Board[x, y]);
```

Comment:
Not every piece has valid moves each turn. We should visualize the pieces that can be moved each turn.

Reply:
This could be handled under a future enhancement if necessary. I won't hold up merging this pull request on this topic.